process/message.go

```diff
@@ -221,8 +221,6 @@ func DecodeMessage(data []byte) (Message, error) {
 		m = &CollectorNode{}
 	case TypeCollectorCluster:
 		m = &CollectorCluster{}
-	case TypeCollectorManifest:
```

Comment:
The type still exists, why remove it here or why not remove both together?

Reply:
Because we use type to send messages but we have different way to encode manifest messages (compress the whole message ) so this func DecodeMessage is not suitable for manifest payloads.

Reply:
So why didn't you remove the Type itself?

Reply:
You mean remove type `TypeCollectorManifest` itself ?
I use this type in agent to encode manifest payloads https://github.com/DataDog/datadog-agent/blob/45bf8f76a82f435e337128969d832d40b36889ee/pkg/process/util/api/payload.go#L38-L40
And in writer to write docs :
https://github.com/DataDog/dd-go/blob/a22f2ed409094c5925d4c3135f259fc0e5ae981b/process/store/persister.go#L202-L205

Reply:
In any case, if you are using it somewhere else in our pipeline you should give some rationality why this change is needed and/or not needed especially since its an API change and difficult to know the why on this change from the code changes alone 👍

Reply:
I add one line comment beside manifest type to make it clear https://github.com/DataDog/agent-payload/pull/158/commits/e01f9b159a4a916f1c8c282f2d02d485bf798d92